specifications/M1/verifiable.presentation.protocol.md

```diff
@@ -0,0 +1,209 @@
+provide the ability to selectively restrict access to resources. The access control mechanism must support the scope
+restrictions defined in [Section 3.1](#31-access-scopes). Implementations may support additional restriction methods,
+including requiring the requester to present its own VPs.
+
+# 4. Resolution API
+
+If a client is not authorized for an endpoint request, the Credential Service may
+return `401 Not Authorized`, `403 Forbidden`, or `404 Not Found`.
+
+## 4.1. Presentations
+
+### 4.1.1. Query For Presentations
+
+Presentations can be queried by POSTing a message to the query endpoint:
+
+`POST /presentations/query`
+
+```json
+{
+  "@context": [
+    "https://w3id.org/catenax/2023/cs/v1",
+    "https://identity.foundation/presentation-exchange/submission/v1"
+  ],
+  "@type": "Query",
+  "presentation_definition": "...",
+  "scope": []
+}
+```
+
+A `Query` MUST contain either a `presentation_definition` or a `scope` parameter. It is an error to contain both.
+
+### 4.1.1.1. Presentation Definitions
+
+Implementations MAY support the `presentation_definition` parameter. If they do not, they MUST
+return `501 Not Implemented`. The `presentation_definition` parameter contains a valid `Presentation Definition`
+according to
+the [Presentation Exchange Specification](https://identity.foundation/presentation-exchange/spec/v2.0.0/#presentation-definition).
+The CS may require an authorization token to authorize the request and uses the presentation definition to return a set
+of matching VPs in the format specified by the definition.
+
+### 4.1.1.2. Scopes
+
+Implementations MAY support requesting presentation of Verifiable Credentials using OAuth 2.0 OAuth 2.0 scope values.
+
+Such a scope value MUST be an alias for a well-defined Presentation Definition. The specific scope values, and the
+mapping between a certain scope value and the respective Presentation Definition is out of scope of this specification.
+
+### 4.1.1.3. Response
+
+The response type of a presentation query is defined by
+the [OpenID for Verifiable Presentations specifications](https://openid.net/specs/openid-4-verifiable-presentations-1_0.html#section-6.1).
+
+> TODO: This needs to be elaborated
+> TODO: Elaborate error responses
+
+# 5. Storage API
+
+VCs can be written to the Credential Service by POSTing an array of [Verifiable Credentials]() to
+the `credentials` endpoint
+
+`POST /credentials`
+
+If successful, credentials will be created or updated and a `200 OK` returned.
+
+# 6. CS Endpoint Resolution through DID Documents
+
+Different methods may be used by a Relying Party to resolve the Credential Service for a client. One way is
+through DID documents. If a DID document is used, the client `DID` document MUST contain at least
+one [service entry](https://www.w3.org/TR/did-core/#services) of type `CredentialService`:
+
+```
+{
+  "service": [
+    {
+      "id":"did:example:123#identity-hub",
```

Comment:
If we use a service we have to ensure the Authentication for this service has the same #UUID at the end so if we use a service we have to define more stuff in the DID document

Reply:
Can you explain, not sure I follow?

Reply:
![image](https://github.com/eclipse-tractusx/ssi-docu/assets/104348964/05217f90-f931-4508-b98c-e50e42bf51b7)
If we use the Service ID, there has to be an equivalent id in the authentication method

Reply:
*verification method sorry

Reply:
I think this may be another issue. The service endpoint is only for resolving the Credential Service. The verification method would be used to verify a self-issued token.